dvc/external_repo.py

```diff
@@ -102,19 +125,29 @@ def __init__(
         uninitialized=False,
         **kwargs,
     ):
-        super().__init__(
-            root_dir, scm=scm, rev=rev, uninitialized=uninitialized
-        )
-
         self.url = url
         self.for_write = for_write
-        self.cache_dir = cache_dir or self._get_cache_dir()
-        self.cache_types = cache_types
-
-        self._setup_cache(self)
-        self._fix_upstream(self)
         self.tree_confs = kwargs
 
+        self._cache_config = {
+            "cache": {
+                "dir": cache_dir or self._get_cache_dir(),
```

Comment:
Btw, we are and have been using the same cache dir for the main repo and its subrepos. Looks intentional, but just want to put a mind note...